File: CicloAndroid/app/src/main/java/com/cursos/android/cicloandroid/MainActivity.java
package com.cursos.android.cicloandroid;

import androidx.appcompat.app.AppCompatActivity;

import android.media.MediaPlayer;
import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    private MediaPlayer mediaPlayer = null;
    private int position = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Log.i("Ciclo de Vida", "OnCreate");
    }

    @Override
    protected void onStart(){
        super.onStart();
        mediaPlayer = MediaPlayer.create(this,R.raw.audio);
        mediaPlayer.start();
        Log.i("Ciclo de Vida", "onStart");
    }

    @Override
    protected void onResume(){
        super.onResume();
        if(mediaPlayer != null){
            mediaPlayer.seekTo(position);
            mediaPlayer.start();
        }
        Log.i("Ciclo de Vida", "onResume");
    }

    @Override
    protected void onPause(){
        super.onPause();
        if(mediaPlayer!= null){
            mediaPlayer.pause();
            position = mediaPlayer.getCurrentPosition();
        }
        Log.i("Ciclo de Vida", "onPause");

    }

    @Override
    protected void onStop(){
        super.onStop();
        if(mediaPlayer!= null) {
            mediaPlayer.stop();
            mediaPlayer.release();
            mediaPlayer = null;
        }
        Log.i("Ciclo de Vida", "onStop");

    }

    @Override
    protected void onRestart(){
        super.onRestart();
        Log.i("Ciclo de Vida","onRestart");

    }

    @Override
    protected void onDestroy(){
        super.onDestroy();
        Log.i("Ciclo de Vida", "On Destroy");
    }
}

File: Ciclo_De_Vida_Android/app/src/main/java/com/example/ciclo_de_vida_android/MainActivity.kt
package com.example.ciclo_de_vida_android

import android.media.MediaPlayer
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class MainActivity : AppCompatActivity() {

    private var mediaPlayer: MediaPlayer? = null
    private var position: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        Log.i("CiclodeVida", "OnCreate")
    }

    override fun onStart(){
        super.onStart()
        mediaPlayer = MediaPlayer.create(this, R.raw.music_1)
        mediaPlayer?.start()
        Log.i("CiclodeVida","OnStart")
    }

    override fun onResume() {
        super.onResume()
        mediaPlayer?.seekTo(position)
        mediaPlayer?.start()

        Log.i("CiclodeVida","OnResume")
    }

    override fun onPause() {
        super.onPause()
        mediaPlayer?.pause()
        if (mediaPlayer != null)
        position = mediaPlayer!!.currentPosition
        Log.i("CiclodeVida","OnPause")
    }

    override fun onStop() {
        super.onStop()
        mediaPlayer?.stop()
        mediaPlayer?.release()
        mediaPlayer = null
        Log.i("CiclodeVida","OnStop")
    }

    override fun onRestart() {
        super.onRestart()

        Log.i("CiclodeVida","OnRestart")
    }

    override fun onDestroy() {
        super.onDestroy()

        Log.i("CiclodeVida","onDestroy")
    }
}